python_modules/dagster/dagster/_core/definitions/decorators/asset_decorator.py

```diff
@@ -444,7 +445,7 @@ def inner(fn: Callable[..., Any]) -> AssetsDefinition:
 
         arg_resource_keys = {arg.name for arg in get_resource_args(fn)}
         check.param_invariant(
-            len(required_resource_keys or []) == 0 or len(arg_resource_keys) == 0,
```

Comment:
this needs to be fixed, issue w/ passing `resource_defs` alongside encoding resources as params